eclair-core/src/main/scala/fr/acinq/eclair/channel/Register.scala

```diff
@@ -48,8 +48,13 @@ class Register extends Actor with ActorLogging {
     case ChannelIdAssigned(channel, remoteNodeId, temporaryChannelId, channelId) =>
       context become main(channels + (channelId -> channel) - temporaryChannelId, shortIds, channelsTo + (channelId -> remoteNodeId) - temporaryChannelId)
 
-    case ShortChannelIdAssigned(_, channelId, shortChannelId, _) =>
-      context become main(channels, shortIds + (shortChannelId -> channelId), channelsTo)
+    case ShortChannelIdAssigned(_, channelId, shortChannelId, previousShortChannelId) =>
```

Comment:
```suggestion
    case ShortChannelIdAssigned(_, channelId, shortChannelId, previousShortChannelId_opt) =>
```

to prevent shadowing